File: src/UslugiRyadom.Api/Entities/Order.cs
using UslugiRyadom.Api.Enums;

namespace UslugiRyadom.Api.Entities;

public class Order
{
    public Guid Id { get; set; }

    public Guid ClientId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public Guid CategoryId { get; set; }

    public string City { get; set; } = string.Empty;

    public string District { get; set; } = string.Empty;

    public string? Street { get; set; }

    public string? House { get; set; }

    public string? Landmark { get; set; }

    public decimal Price { get; set; }

    public PaymentType PaymentType { get; set; }

    public OrderStatus Status { get; set; } = OrderStatus.New;

    public Guid? MasterId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ApplicationUser Client { get; set; } = null!;

    public ApplicationUser? Master { get; set; }

    public Category Category { get; set; } = null!;
}

File: src/UslugiRyadom.Web/src/types/order.ts
import type { Address } from '@/types/common';

export type OrderStatus = 'new' | 'accepted' | 'inProgress' | 'completed' | 'cancelled';
export type PaymentType = 'cash' | 'card';

export interface CategorySummary {
  id: string;
  name: string;
  slug: string;
}

export interface UserSummary {
  id: string;
  fullName: string;
  phone?: string;
}

export interface OrderListItem {
  id: string;
  title: string;
  price: number;
  status: OrderStatus;
  paymentType: PaymentType;
  createdAt: string;
  address: Address;
  category: CategorySummary;
  client: UserSummary;
  master?: UserSummary | null;
}

export interface Order extends OrderListItem {
  description: string;
  updatedAt: string;
}

export interface OrderFilters {
  categoryId?: string;
  city?: string;
  district?: string;
  status?: OrderStatus;
  clientId?: string;
  masterId?: string;
  pageNumber?: number;
  pageSize?: number;
}

export interface CreateOrderRequest extends Address {
  title: string;
  description: string;
  categoryId: string;
  city: string;
  district: string;
  price: number;
  paymentType: PaymentType;
}
